File: FlexBP/app/src/main/java/vlac/navi/flexbp/MainActivity.java
package vlac.navi.flexbp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        // this is just for the first time
        // also it is executed when occurs a rotation
        // even in screen size
        System.out.println("onCreate");
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    @Override
    protected void onStart() {
        System.out.println("onStart");
        super.onStart();
    }

    @Override
    protected void onResume() {
        // this happens when the view is seeing by user
        System.out.println("onResume");
        super.onResume();
    }

    @Override
    protected void onPause() {
        // it is just happens in multiviews
        System.out.println("onPause");
        super.onPause();
    }

    @Override
    protected void onStop() {
        // it just happens when another activity covers all screen
        System.out.println("onPause");
        super.onStop();
    }

    @Override
    protected void onRestart() {
        System.out.println("onRestart");
        super.onRestart();
    }

    @Override
    protected void onDestroy() {
        System.out.println("onDestroy");
        super.onDestroy();
    }

    public void signIn(View view) {
        Intent intent = new Intent(this, DisplayMessageActivity.class);
        EditText txtUsername = (EditText) findViewById(R.id.editTextTextPersonName);
        EditText txtPassword = (EditText) findViewById(R.id.editTextTextPassword);
        String username = txtUsername.getText().toString();
        String password = txtPassword.getText().toString();
        intent.putExtra("vlac.navi.flexbp.USER", username);
        intent.putExtra("vlac.navi.flexbp.PASSWORD", password);
        startActivity(intent);
        // FLAC_ACTIVITY_CLEAR_TOP - Delete all activities over this
        // clearTaskOnLaunch - all stack will be deleted but the root is not
    }

    public void signUp(View view) {}
}

File: DiceRoller/app/src/main/java/vlac/navi/diceroller/MainActivity.kt
package vlac.navi.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    lateinit var diceImg: ImageView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollBtn: Button = findViewById(R.id.roll_btn)
        rollBtn.text = "Let's Roll"
        rollBtn.setOnClickListener {
            rollDice()
        }
        this.diceImg = findViewById(R.id.dice_img)
    }

    private fun rollDice() {
        val diceResource = when(Random.nextInt(1,7)) {
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }
        this.diceImg.setImageResource(diceResource)
    }
}